code/game/machinery/regenerator.dm

```diff
@@ -19,15 +28,80 @@
 		if(!("neutral" in L.faction)) // Enemy spotted
 			regen_amt *= 0.5
 			break
+	if(burst)
+		regen_amt *= 7.5
+		burst = FALSE
+		cooldown = TRUE
+		addtimer(CALLBACK(src, .proc/Reset_Cooldown), 30 SECONDS)
```

Comment:
Instead you should use variable cooldowns, see how abnormalities do this.

Reply:
This one is SPECIFICALLY for turning off the machine for 30 seconds after the "Burst" function has been used. The others use a timer but I'll see what I can do to standardize the other timers.